bootstrap/main.go

```diff
@@ -0,0 +1,209 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+
+	flag "github.com/spf13/pflag"
+)
+
+func printUsage(flags *flag.FlagSet) {
+	log.Println("Flags")
+	log.Println(flags.FlagUsages())
+}
+
+// ParseArgs parses the command line to supported arguments as a pflag.FlagSet
+// arguments should be generally be os.Args[1:]
+func ParseArgs(arguments []string) *flag.FlagSet {
+	args := flag.NewFlagSet("bootstrap", flag.ExitOnError)
+	// used to mimic required=true
+	requiredFlags := []string{"job"}
+
+	// add all of the args from parse_args in jenkins/bootstrap.py
+	args.String("root", ".", "Root dir to work with")
+
+	// NOTE: jenkins/bootstrap.py technically used a float for this arg
+	// when parsing but only ever used the arg as an integer number
+	// of timeout minutes.
+	// int32 makes more sense and will let use time.Minute * timeout
+	args.Int32("timeout", 0, "Timeout in minutes if set")
+
+	// TODO(bentheelder): remove these?
+	args.String("pull", "", "Deprecated, use --repo=k8s.io/foo=master:abcd,12:ef12,45:ff65")
+	args.String("branch", "", "Deprecated, use --repo=k8s.io/foo=master")
+
+	args.StringArray("repo", []string{},
+		"Fetch the specified repositories, with the first one considered primary")
+
+	args.Bool("bare", false, "Do not check out a repository")
+	args.Lookup("bare").NoOptDefVal = "true" // allows using --bare
+
+	// NOTE: this arg is required (set above)
+	args.String("job", "", "Name of the job to run")
+
+	args.String("upload", "", "Upload results here if set, requires --service-account")
+	args.String("service-account", "", "Activate and use path/to/service-account.json if set.")
+	args.String("ssh", "", "Use the ssh key to fetch the repository instead of https if set.")
+	args.String("git-cache", "", "Location of the git cache.")
+
+	args.Bool("clean", false, "Clean the git repo before running tests.")
+	args.Lookup("clean").NoOptDefVal = "true" // allows using --clean
+
+	// parse then check required flags
+	args.Parse(arguments)
+	for _, arg := range requiredFlags {
+		if flag := args.Lookup(arg); !flag.Changed {
+			log.Printf("Flag '--%s' is required but was not set!\n", flag.Name)
+			printUsage(args)
+			os.Exit(-1)
+		}
+	}
+
+	// validate args
+	repo, _ := args.GetStringArray("repo")
+	bare, _ := args.GetBool("bare")
+	if bare == (len(repo) != 0) {
+		log.Printf("Expected --repo xor --bare, Got: --repo=%v, --bare=%v\n", repo, bare)
+		printUsage(args)
+		os.Exit(-1)
+	}
+	if len(repo) != 0 {
```

Comment:
Drop support for `--pull` and `--branch` and all related logic

Reply:
done.